File: My-sky-main/src/models/Event.ts
export interface Event {
  id: string;
  title: string;
  description: string;
  photos: string[];
  location: {
    latitude: number;
    longitude: number;
    address: string;
  };
  startDateTime: Date;
  endDateTime: Date;
  createdBy: string; // ユーザーID
  circle?: string; // サークルID（任意）
  participants: {
    userId: string;
    status: 'attending' | 'maybe' | 'declined';
  }[];
  maxParticipants?: number;
  createdAt: Date;
  updatedAt: Date;
}

export interface EventCreation {
  title: string;
  description: string;
  photos: string[];
  location: {
    latitude: number;
    longitude: number;
    address: string;
  };
  startDateTime: Date;
  endDateTime: Date;
  circle?: string; // サークルID（任意）
  maxParticipants?: number;
}

export type ParticipationStatus = 'attending' | 'maybe' | 'declined'; 

File: mysky-api/src/main/java/com/mysky/api/entity/Event.java
package com.mysky.api.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "events")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Event {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    @Column(length = 2000)
    private String description;

    private String category;
    private String imageUrl;

    @Column(nullable = false)
    private LocalDateTime startTime;

    private LocalDateTime endTime;

    private Double latitude;
    private Double longitude;
    private String location;
    private String venue;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "creator_id", nullable = false)
    private User creator;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "circle_id")
    private Circle circle;

    @Enumerated(EnumType.STRING)
    @Builder.Default
    private EventStatus status = EventStatus.UPCOMING;

    @ManyToMany
    @JoinTable(
        name = "event_attendees",
        joinColumns = @JoinColumn(name = "event_id"),
        inverseJoinColumns = @JoinColumn(name = "user_id")
    )
    @Builder.Default
    private Set<User> attendees = new HashSet<>();

    private Integer maxAttendees;

    @CreationTimestamp
    private LocalDateTime createdAt;

    @UpdateTimestamp
    private LocalDateTime updatedAt;

    public enum EventStatus {
        UPCOMING, ONGOING, COMPLETED, CANCELLED
    }
}
